Project/Assets/Prairie/Framework/Script/Twine/TwineNode.cs

```diff
@@ -73,149 +79,157 @@ private void Start()
-            }
-            if (this.isConditionNode)
-            {
-                // get the $color value from global list
-                // check the platform name by check $color:platform pair stored in condition node
-                // check child node name to match platform name
-                // activate childnode
-                this.ActivateChildAtIndex(0);
-                Debug.Log("This is a Condition Node: " + this.name);
             }
         }
-    }
-
-    public void OnGUI()
-    {
-        if (isDecisionNode)
+    
+        public void OnGUI()
         {
-            int frameWidth = Math.Min(Screen.width / 3, 500);
-            int frameHeight = Math.Min(Screen.height / 2, 350);
-            int horizontalAlign = (Screen.width - frameWidth) / 2;
-            int verticalAlign = Screen.height - frameHeight;
-
-            Rect frame = new Rect(horizontalAlign, verticalAlign, frameWidth, frameHeight);
-
-            GUI.BeginGroup(frame);
-            GUIStyle style = new GUIStyle(GUI.skin.box);
-            style.normal.textColor = Color.white;
-            style.wordWrap = true;
-            style.fixedWidth = frameWidth;
-            GUILayout.Box(this.content, style);
-            for (int index = 0; index < this.validLinkNames.Length; index++)
-            {
-                if (GUILayout.Button(this.validLinkNames[index]))
-                {
-                    this.ActivateChildAtIndex(index);
+            if ((fanfold) && (allMinimized == false)) {
+                float frameWidth = Math.Min(Screen.width / 5, 200);
+                float frameHeight = 80;
+                if (TwineNodeList.Count() > 0){
+                    frameHeight = Math.Min(Screen.height / (TwineNodeList.Count()), 80);
                 }
+                int index = TwineNodeList.IndexOf(this);
+                Rect frame = new Rect (10, 10+index*80, frameWidth, frameHeight);
+                GUIStyle style = new GUIStyle (GUI.skin.box);
+                style.wordWrap = true;
+                style.fixedWidth = frameWidth-10;
+                GUI.BeginGroup(new Rect(20, 10+index*frameHeight, frameWidth + 10, frameHeight));
+                scrollPosition = GUILayout.BeginScrollView(scrollPosition, GUILayout.Width(frameWidth + 10), GUILayout.Height(frameHeight));
+                content = content.TrimEnd();
+                GUILayout.Label(new GUIContent(this.content), style, GUILayout.Width(frameWidth - 20), GUILayout.ExpandHeight(true));
+                
+                if (isDecisionNode) {
+                    GUIStyle decisionHintStyle = new GUIStyle (style);
+                    decisionHintStyle.fontStyle = FontStyle.BoldAndItalic;
+
+                    if (!isOptionsGuiOpen) {
+                        GUILayout.Box ("Press TAB to progress in the story...", decisionHintStyle);
+                    } else {
+                        GUILayout.Box ("Press TAB to scroll, E to close, ENTER to choose", decisionHintStyle);
```

Comment:
Same "'E' to close" question as in the previous PL. Is it tested with the cursor pointing to an object with carrying script attached? What will happen when that object both have decision node triggered and carry script attached?